### PR TITLE
WIP: Batch-correction with `limma`

### PR DESCRIPTION
**Pull request by cansavvy**
## Purpose:
This PR addresses #24. This was previously attempted with a ComBat example, but a good dataset was not found that illustrated ComBat's correction. In this revised example. I'm using `limma`'s `removeBatchEffect` on a smashed dataset of 4 AML datasets (where two are RNA-seq, and two are hgu133plus2). 

What is still to be determined is how to this dataset should be retrieved. It is too big to store on GitHub, even after compression. 

## Pull Request Check List:
* [x] Run a linter
* [x] Set the seed (if applicable)
* [x] Comments and/or documentation up to date
* [x] Double check your paths
* [x] Check headline numbering
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
@cansav09 what linter did you settle on?

**Comment by cansav09** [comment]
I’ve been using `lintr`.

**Comment by jaclyn-taroni** [comment]
I would have expected any linter that enforces our selected style guide to flag variable names with `.` rather than `_` in them and to possibly catch the spacing issue in https://github.com/AlexsLemonade/refinebio-examples/pull/82/files#diff-ca67dcdb1a515534dd1cfe7d11e268aaR90

**Comment by cansav09** [comment]
It didn’t bring those things up, but it’s also very possible I haven’t set it up right. It brought up other spacing issues though. Which linter do you use?

**Comment by jaclyn-taroni** [comment]
I generally don’t use one and was hoping to pick one that works for both of us. That’s why I’m asking. We might want to check this out in addition: https://github.com/r-lib/styler

Looks pretty customizable?

**Comment by cansav09** [comment]
Looks worth looking into. I also haven’t checked out all the capabilities of `lintr` or any other lingers for that matter. I will do some more investigating and testing of  these tomorrow.

**Comment by jaclyn-taroni** [comment]
When you go back through @cansav09, please let me know if you need me to clarify anything that is ambiguous. It may be helpful to comment stating the changes you intend to make/the direction you intend to go in to make sure we're on the same page.

**Comment by cansav09** [comment]
Yes, I can address the other changes we want made for now; but probably for the final merging we will have to hold off until those other issues you mentioned above get addressed. (As well as the determination of where the data files should go).

File: batch-correction/batch_correction.Rmd [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+### General requirements for performing batch correction:   
+
+#### 1) Have adequate sample sizes per batch  
+For batch correction to work you need adequate sample size (> 10 at least) for 
+each batch group.
+In this example, we will consider the original datasets and platforms as 
+possible batch variables to consider correcting by. 
+
+#### 2) Cannot have confounds between batch and other biological variables
```
[the remark this answers is not in the document]
Comment (cansav09) [reply batch-correction/batch_correction.Rmd]: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4072626/

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply batch-correction/batch_correction.Rmd]: Sorry that was not clear.  The question was specifically about the phrase "have confounds."

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply batch-correction/batch_correction.Rmd]: https://github.com/AlexsLemonade/refinebio-examples/pull/82/commits/21fd0ee3d87d982e4871da16ac9fc46119e99718 👍 🤘  (trying to log into all the things on my phone and therefore staring at my email  👀 )

**Comment by cansav09** [comment]
>AlexsLemonade/refinebio#1299, AlexsLemonade/refinebio#1296, AlexsLemonade/refinebio-frontend#633. 

Note that these issues are no longer blocking this.

**Comment by jaclyn-taroni** [comment]
I'm going to close this since these materials are getting an overhaul. We can always revisit this material.

**Pull request closed without merging**